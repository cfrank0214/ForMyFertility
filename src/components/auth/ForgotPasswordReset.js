import React, { Component } from 'react';
import {
    Container,
    Body,
    Button,
    Form,
    Text,
    Header,
    Content,
    Item,
    Input,
    Label
  } from 'native-base';
import { Col, Row, Grid } from 'react-native-easy-grid';
import { Auth, Logger } from 'aws-amplify';

const logger = new Logger('ForgotPasswordReset');

export default class ForgotPasswordReset extends Component {
  constructor(props) {
    super(props);
    this.submit = this.submit.bind(this);
    this.changeState = this.changeState.bind(this);
    this.state = { error: null }
  }

  changeState(state, data) {
    const { onStateChange } = this.props;
    if (onStateChange) {
      onStateChange(state, data);
    }
  }

  submit() {
    const username = this.props.authData;
    if (!username) {
      this.setState({ error: 'missing username' });
      return;
    }

    const { code, password } = this.state;
    logger.info('reset password for ' + username);
    Auth.forgotPasswordSubmit(username, code, password)
      .then(data => this.submitSuccess(username, data))
      .catch(err => this.handleError(err));
  }

  submitSuccess(username, data) {
    logger.info('forgot password reset success for ' + username, data);
    this.changeState('signIn', username);
  }

  handleError(err) {
    logger.info('forgot password reset error', err);
    this.setState({ error: err.message || err });
  }

  render() {
    const { authState } = this.props;
    if (authState !== 'forgotPasswordReset') { return null; }

    const { error } = this.state;

    return (
        <Content>
          <Grid>
            <Form  >
              <Item stackedLabel style={{ marginTop: 30 }}>
                <Label>Code</Label>
                <Input 
                  type="text"
                  onChangeText={(code) => this.setState({ code })}
                  autoFocus
                />
              </Item>
              <Item stackedLabel last>
                <Label>Password</Label>
                <Input 
                  type="password"
                  autoCapitalize="none"
                  onChangeText={(password) => this.setState({ password })}
                  autoFocus
                />
              </Item>
              <Button block success onPress={() => { this.submit() }}>
                  <Text>
                  Reset password
                  </Text>
              </Button>
              {error &&
                <Text>{error}</Text>
              }
              <Row>
              
                <Button transparent onPress={() => this.changeState('forgotPassword')} >
                  <Text>
                  Back to forgot password
                  </Text>
                </Button>
              </Row>
            </Form>
          </Grid>
        </Content>
    )
  }
}