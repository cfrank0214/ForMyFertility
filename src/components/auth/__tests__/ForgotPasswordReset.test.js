/**
 * @jest-environment jsdom
 */
import React from 'react'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ForgotPasswordReset from '../ForgotPasswordReset';
import {
  Button,
  Form,
  Text,
  Header,
  Content,
  Item,
  Input,
  Label
} from 'native-base';
import { Col, Row, Grid } from 'react-native-easy-grid';

describe('>>>ForgotPasswordReset --- Shallow Render REACT COMPONENTS', () => {
  let wrapper

  beforeEach(() => {

    const props = {
      inputs: jest.fn(),
      state: jest.fn(),
      authState: 'forgotPasswordReset'
    }

    wrapper = shallow(<ForgotPasswordReset {...props} />)

  })

  it('renders correctly', () => {
    const tree = renderer.create(
      <ForgotPasswordReset />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('+++ render the DUMB component', () => {
    expect(wrapper.length).toEqual(1)
  });
  it('The <Button /> component exists', () => {
    expect(wrapper.find(Button).exists()).toBe(true);
  });
  it('contains 2 <Button /> components', () => {
    expect(wrapper.find(Button).length).toBe(2);
  });
  it('contains 1 <Form /> component', () => {
    expect(wrapper.find(Form).length).toBe(1);
  });
  it('contains 2 <Input /> component', () => {
    expect(wrapper.find(Input).length).toBe(2);
  });
  it('contains 1 <Row /> component', () => {
    expect(wrapper.find(Row).length).toBe(1);
  });
});