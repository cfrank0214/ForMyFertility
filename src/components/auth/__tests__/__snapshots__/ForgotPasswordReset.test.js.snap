// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>ForgotPasswordReset --- Shallow Render REACT COMPONENTS renders correctly 1`] = `null`;
